tests/icon-api.js

```diff
@@ -0,0 +1,252 @@
+    iconId: iconDoc.docId,
+    size: 'large',
+  })
+  t.alike(icon, expectedLargeIcon)
+})
+
+test('icon api - create and get with variants, choosing the variant with more matching criteria', async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+      {
+        size: 'medium',
+        pixelDensity: 1,
+        mimeType: 'image/svg+xml',
+        blob: Buffer.from(expectedMediumIcon),
+      },
+      {
+        size: 'large',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedLargeIcon),
+      },
+    ],
+  })
+
+  const { icon } = await iconApi[kGetIcon]({
+    iconId: iconDoc.docId,
+    size: 'large',
+    pixelDensity: 1,
+    mimeType: 'image/png',
+  })
+  t.alike(icon, expectedLargeIcon)
+})
+
+test('icon api - create and get with variants, choosing the first variant with the first best score', async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+      {
+        size: 'medium',
+        pixelDensity: 1,
+        mimeType: 'image/svg+xml',
+        blob: Buffer.from(expectedMediumIcon),
+      },
+      {
+        size: 'large',
+        pixelDensity: 2,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedLargeIcon),
+      },
+    ],
+  })
+
+  const { icon } = await iconApi[kGetIcon]({
+    iconId: iconDoc.docId,
+    size: 'large',
+    pixelDensity: 1,
+    mimeType: 'image/svg+xml',
+  })
+  t.alike(icon, expectedMediumIcon)
+})
+
+test(`icon api - getIconUrl, test matching url`, async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+    ],
+  })
+
+  const size = 'small'
+  const pixelDensity = 1
+  const expectedUrl = `/${projectId}/${iconDoc.docId}/${size}/${pixelDensity}`
+  const url = await iconApi[kGetIconUrl]({
+    iconId: iconDoc.docId,
+    size: 'small',
+    pixelDensity: 1,
+  })
+  t.is(url, expectedUrl)
+})
+
+test(`icon api - getBestVariant`, async (t) => {
+  const variants = [
+    {
+      size: 'small',
+      pixelDensity: 1,
+      mimeType: 'image/png',
+      blob: Buffer.from(expectedSmallIcon),
+    },
+    {
+      size: 'medium',
+      pixelDensity: 1,
+      mimeType: 'image/svg+xml',
+      blob: Buffer.from(expectedMediumIcon),
+    },
+    {
+      size: 'large',
+      pixelDensity: 2,
+      mimeType: 'image/png',
+      blob: Buffer.from(expectedLargeIcon),
```

Comment:
these should be `blobVersionId` instead of `blob` based on the `kGetBestVariant` signature